### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

**Comment by ksheremet** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover/pull-request/81/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover)

**Comment by dotdoom** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover/pull-request/81/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover)

File: main.go [hunk 1]
```diff
@@ -104,11 +111,35 @@ func (s *server) ReadEncoders(ctx context.Context,
...
+	if err != nil {
+		log.Println("Failed to listen:", err)
+	}
+	opts, err := setServerOptions()
+	if err != nil {
+		log.Println("Failed to setup Server Options:", err)
+	}
+	s := grpc.NewServer(opts...)
+	pb.RegisterRoverServiceServer(s, &server{})
+	err = s.Serve(lis)
```
Comment (dotdoom) [inline main.go]: Why not just `return s.Serve(lis)`?

File: main.go [hunk 2]
```diff
@@ -104,11 +111,35 @@ func (s *server) ReadEncoders(ctx context.Context,
...
+		}
+		return []grpc.ServerOption{grpc.Creds(creds)}, nil
+	}
+	return opts, nil
+}
+
+func startServer() error {
+	lis, err := net.Listen("tcp", *laddr)
+	if err != nil {
+		log.Println("Failed to listen:", err)
```
Comment (dotdoom) [inline main.go:129]: Is there any good reason to proceed if `err` is not `nil`?

File: main.go [hunk 3]
```diff
@@ -104,11 +111,35 @@ func (s *server) ReadEncoders(ctx context.Context,
...
+	}
+	opts, err := setServerOptions()
+	if err != nil {
+		log.Println("Failed to setup Server Options:", err)
```
Comment (dotdoom) [inline main.go:134]: ditto

**Comment by dotdoom** [comment]
Approved!

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover/pull-request/81/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover)

**Pull request merged**